src/controller_examples/rabbitmq_controller/proof/liveness/helper_invariants/owner_ref.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use super::invariants::*;
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::{
+    api_method::*, common::*, config_map::*, dynamic::*, owner_reference::*, resource::*,
+    stateful_set::*,
+};
+use crate::kubernetes_cluster::spec::{
+    builtin_controllers::types::{built_in_controller_req_msg, BuiltinControllerChoice},
+    cluster::*,
+    cluster_state_machine::Step,
+    controller::common::{controller_req_msg, ControllerActionInput, ControllerStep},
+    message::*,
+};
+use crate::rabbitmq_controller::{
+    common::*,
+    proof::common::*,
+    spec::{rabbitmqcluster::*, reconciler::*},
+};
+use crate::temporal_logic::{defs::*, rules::*};
+use vstd::prelude::*;
+
+verus! {
+
+spec fn sts_key_not_exists_or_delete_msg_in_flight_or_sts_has_current_ref(rabbitmq: RabbitmqClusterView) -> StatePred<RMQCluster> {
```

Comment:
Why not just say if key exists then either exists delete msg or owner ref points to the current cr?

Reply:
I find it makes more sense to say `key_exists /\ owner_ref != [current_cr] ==> delete_msg_in_flight`.

Reply:
Then you can go that way